specs/system_config.md

```diff
@@ -0,0 +1,68 @@
+# System Config
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [System config contents](#system-config-contents)
+  - [`batcherHash` (`bytes32`)](#batcherhash-bytes32)
+  - [`l1FeeOverhead` and `l1FeeScalar` (`uint256,uint256`)](#l1feeoverhead-and-l1feescalar-uint256uint256)
+- [Writing the system config](#writing-the-system-config)
+- [Reading the system config](#reading-the-system-config)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+The `SystemConfig` is a contract on L1 that can emit rollup configuration changes as log events.
+The rollup [block derivation process](./derivation.md) picks up on these log events, and applies the changes.
+
+## System config contents
+
+### `batcherHash` (`bytes32`)
+
+A versioned hash of the current authorized batcher sender(s), to rotate keys as batch-submitter.
+The first byte identifies the version.
+
+Version `0` embeds the current batch submitter ethereum address (`bytes20`) in the last 20 bytes of the versioned hash.
+
+In the future this versioned hash may become a commitment to a more extensive configuration,
+to enable more extensive redundancy and/or rotation configurations.
+
+### `l1FeeOverhead` and `l1FeeScalar` (`uint256,uint256`)
+
+The L1 fee parameters, also known as Gas Price Oracle (GPO) parameters,
+are updated in conjunction and apply new L1 costs to the L2 transactions.
+
+## Writing the system config
+
+The `SystemConfig` contract applies authentication to all writing contract functions,
+the configuration management can be configured to be any type of ethereum account or contract.
+
+On write, an event is emitted for the change to be picked up by the L2 system,
+and a copy of the new written configuration variable is retained in L1 state to read with L1 contracts.
+
+## Reading the system config
+
+A rollup node initializes its derivation process by finding a starting point based on its past L2 chain:
+
+- When started from L2 genesis, the initial system configuration is retrieved from the rollup chain configuration.
+- When started from an existing L2 chain, a previously included L1 block is determined as derivation starting point,
+  and the system config can thus be retrieved from the last L2 block that referenced the L1 block as L1 origin:
+  - `batcherHash`, `l1FeeOverhead` and `l1FeeScalar` are retrieved from the L1 block info transaction.
+  - other future variables may also be retrieved from other contents of the L2 block, such as the header.
+
+After preparing the initial system configuration for the given L1 starting input,
+with each new L1 block the system configuration is updated by copying the configuration
+and then applying any emitted log events from the L1 `SystemConfig` contract:
```

Comment:
Need to specify that the system config is updated by the receipts of a block before scanning the block's transactions & the GPO parameters take effect in the first L2 block  whose L1 origin is the block in which it was changed.